### PR TITLE
Fix for prediction Motor Efficiency curves

### PR DESCRIPTION
**Pull request by e-wai**
In this PR:
+ Move the motor efficiency script and data to dynamics folder
+ Revision of motor efficiency file that @KristinChengWu had written up -> mostly cleaning up functions, moving around initializations, combining graphs into one figure
+ Investigation around why the HI coil looked so weird: fit speed and torque to a curve, then generated points to see the line; solution was removing that one outlier in HI

(This is pretty decent as a prediction, but need to discuss what this looks like in race time)

I cleaned up the graph titles, etc. later but for reference:
Before outlier:
before_outlier
After outlier:
after_outlier

### CODE REVIEW

File: dynamics/motor_efficiency/motor_efficiency.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        self.speed_k = regr.coef_[0]
+        self.torque_k = regr.coef_[1]
+        self.c = regr.intercept_
+
+        # generate fit of speed and torque points to generate data
+        # see speed_torque_func for function used
+        self.rotational_speed = self.dataset['Speed'].values
+        self.torque = self.dataset['Torque'].values
+
+        self.start_pars, cov = curve_fit(f=self.speed_torque_func, xdata=self.rotational_speed, ydata=self.torque)
```
Comment (ufshk) [inline dynamics/motor_efficiency/motor_efficiency.py:46]: where does cov get used, it doesn't seem to be an attribute for the class.

File: dynamics/motor_efficiency/motor_efficiency.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    def graph_generated_points(self):
+        generated_x = np.linspace(self.rotational_speed[0], self.rotational_speed[-1], 100)
+        generated_y = self.speed_torque_func(generated_x, *self.start_pars)
+
+        # If want to show speed vs. power:
+        plt.plot(self.rotational_speed, math.pi / 30 * self.rotational_speed * self.torque, 'r+')
+        plt.plot(generated_x, math.pi / 30 * generated_x * generated_y, 'green')
+        plt.ylabel("Calculated Power (W)")
+        
+        # If want to show speed vs. torque:
```
Comment (ufshk) [inline dynamics/motor_efficiency/motor_efficiency.py:67]: maybe just make this an input to the function? commented out but technically workable lines just seems not good idk

**Review by ufshk — COMMENTED** [review]
I do not have any hard hitting questions.

**Comment by KristinChengWu** [comment]
Thanks emma!

File: dynamics/motor_efficiency/motor_efficiency.py (hunk 1 above)
Comment (e-wai) [reply dynamics/motor_efficiency/motor_efficiency.py:46]: Oh nope, I'll just make it unnamed

**Review by ufshk — APPROVED** [review]
well then

**Review by mdshiozaki — APPROVED** [review]

**Comment by mdshiozaki** [comment]
I feel like i never have comments tbh

**Pull request merged**